spipe/main.c

```diff
@@ -176,16 +179,23 @@ main(int argc, char * argv[])
 	}
 
 	/* Loop until we die. */
-	do {
-		if (events_run()) {
-			warnp("Error running event loop");
-			exit(1);
-		}
-	} while (1);
+	if (events_spin(&conndone)) {
+		warnp("Error running event loop");
+		exit(1);
```

Comment:
Did you want this to be a `goto errN` as well?  Or was the idea that if we die at this point there's no way we can possibly clean everything up, so don't worry about keeping valgrind happy?